### DISCUSSION TITLE
I'm confused by a possible contradiction abt Docker. Maybe I'm mistaken. can enlighten me?

### DISCUSSION THREAD

**Discussion by simkimsia** [silviogutierrez/reactivated#157] [Q&A]
Cool project. Really like your choice of strong opinions. Wish your project came out 3-4 years ago. Of course back then perhaps Nix isn't as advanced as it is today.

Anyway, to my question

I saw https://www.reactivated.io/documentation/deploying/ saying

> Fortunately, just like in local development, we [use Nix](https://www.reactivated.io/documentation/why-nix/) inside a Docker image. It's optimized to be as light as can be, with only the runtime requirements. Review the Dockerfile provided after setup for details.

and then at https://www.reactivated.io/documentation/why-nix/

> Native performance
> You spent thousands of dollars on a machine. It runs quiet. It runs cool. It runs forever on battery. Why throw all that way and run your projects on Docker?

I would definitely like it if i can avoid using Docker. So how do you reconcile these two points? Genuinely curious

**Comment by silviogutierrez** [comment]
So on a platform that natively supports containers, Docker has minimal overhead. This is the case with your production servers most of the time. So if you deploy your Docker image to say, Amazon Fargate or your own VM running Docker, there's no performance hit — or barely any.

On **Mac** specifically, containers are not supported natively. Instead, a Linux VM is launched by Docker Desktop and then it runs the containers there. This alone has a big performance penalty. Moreover, you still need to "sync" files or mount them so you can edit them and see your changes while developing the app. Again, not an issue in production since you're not live editing.

I hope that explains it. Let me know if you need more info.
